contracts/message/libraries/MessageSenderLib.sol

```diff
@@ -68,208 +63,143 @@ library MessageSenderLib {
-     * @param _fee The fee amount to pay to MessageBus.
-     * @return The transfer ID.
-     */
-    function sendMessageWithPegBridgeBurn(
-        MsgDataTypes.BridgeSendType _bridgeSendType,
+    function computePegV1DepositId(
         address _receiver,
         address _token,
         uint256 _amount,
         uint64 _dstChainId,
-        uint64 _nonce,
-        bytes memory _message,
-        address _messageBus,
-        uint256 _fee
-    ) internal returns (bytes32) {
-        address pegBridge;
-        if (_bridgeSendType == MsgDataTypes.BridgeSendType.PegBurn) {
-            pegBridge = IMessageBus(_messageBus).pegBridge();
-        } else {
-            pegBridge = IMessageBus(_messageBus).pegBridgeV2();
-        }
-        IERC20(_token).safeIncreaseAllowance(pegBridge, _amount);
-        bytes32 transferId;
-        if (_bridgeSendType == MsgDataTypes.BridgeSendType.PegBurn) {
-            IPeggedTokenBridge(pegBridge).burn(_token, _amount, _receiver, _nonce);
-            transferId = keccak256(
-                abi.encodePacked(address(this), _token, _amount, _receiver, _nonce, uint64(block.chainid))
+        uint64 _nonce
+    ) internal view returns (bytes32) {
+        return
+            keccak256(
+                abi.encodePacked(address(this), _token, _amount, _dstChainId, _receiver, _nonce, uint64(block.chainid))
             );
-        } else if (_bridgeSendType == MsgDataTypes.BridgeSendType.PegV2Burn) {
-            transferId = IPeggedTokenBridgeV2(pegBridge).burn(_token, _amount, _dstChainId, _receiver, _nonce);
-        } else {
-            // PegV2BurnFrom
-            transferId = IPeggedTokenBridgeV2(pegBridge).burnFrom(_token, _amount, _dstChainId, _receiver, _nonce);
-        }
-        // handle cases where certain tokens do not spend allowance for role-based burn
-        IERC20(_token).safeApprove(pegBridge, 0);
-        if (_message.length > 0) {
-            IMessageBus(_messageBus).sendMessageWithTransfer{value: _fee}(
-                _receiver,
-                _dstChainId,
-                pegBridge,
-                transferId,
-                _message
+    }
+
+    function computePegV1BurnId(
+        address _receiver,
+        address _token,
+        uint256 _amount,
+        uint64 _nonce
+    ) internal view returns (bytes32) {
+        return keccak256(abi.encodePacked(address(this), _token, _amount, _receiver, _nonce, uint64(block.chainid)));
+    }
+
+    function computePegV2Id(
+        address _receiver,
+        address _token,
+        uint256 _amount,
+        uint64 _dstChainId,
+        uint64 _nonce
+    ) internal view returns (bytes32) {
+        return
+            keccak256(
+                abi.encodePacked(
+                    msg.sender,
+                    _token,
+                    _amount,
+                    _dstChainId,
+                    _receiver,
+                    _nonce,
+                    uint64(block.chainid),
+                    address(this)
```

Comment:
This is incorrect, the last address field is the bridge/vault contract address. Anyway, remove this function, use returned Id from the v2 contract when needed.

Reply:
fixed